### ISSUE TITLE
Endpoint types should support Anchor parameter

### ISSUE THREAD

**Issue by damjanaleksiev** [jsplumb/community-edition#143]
I am building an UI were I have multiple boxes that can be connected together. Each box has 1 target and 3 sources, made with the `makeTarget` and `makeSource` methods called upon the same box DOM element. The code looks something like this:

```
jsPlumb.makeTarget($('#box_id'), targetData);
jsPlumb.makeSource($('#box_id'), sourceData1);
jsPlumb.makeSource($('#box_id'), sourceData2);
jsPlumb.makeSource($('#box_id'), sourceData3);
```

And each of the sources is distinguished by the filter property that connects it to a different part of the box:

```
sourceData1.filter = '.filter_1';
sourceData2.filter = '.filter_2';
sourceData3.filter = '.filter_3';
```

Connections are possible to be made only between the sources of one box and the target of another box. The target anchors are `Continuous`.

This arrangement works perfectly for what I need as far as working with the boxes and dragging and dropping connections.

The problem arises when I want to recreate a saved UI from a JSON object. I can recreate the position of the boxes without a problem, but I cannot find a way to recreate the connections to start from the correct source of the 3 available on the box.

I tried to re-connect using the saved uuids, but it turned out the endpoints don't exist yet before the connections are created.
The other method I tried is to connect just the source and target box ids:

```
jsPlumb.connect({ 
    source:"box_id1", 
    target:"box_id2",
});
```

But this never connects the correct source to the target and the connections are not exactly the same as the user made them. If a connection started from source 1, with this method it can start from source 3 for example.

Basically I am asking is it possible to programmatically connect sources and targets before the endpoints are created using the jsPlumb API?

So is it possible to recreate the connections in these circumstances or should I look for other solutions?

Thanks.
